### PR TITLE
Always use port 53 as the dns service port.

### PR DESCRIPTION
**Pull request by abutcher**
Service port should always be 53 while `DNSConfig.BindAddress` port is used for the target port and endpoint port.

@liggitt @smarterclayton

### CODE REVIEW

**Comment by liggitt** [comment]
Add a test to the integration test TestDNS like this:
```
	_, dnsPortString, err := net.SplitHostPort(masterConfig.DNSConfig.BindAddress)
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	dnsPort, err := strconv.Atoi(dnsPortString)
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	kubernetesService, err := client.Services(kapi.NamespaceDefault).Get("kubernetes")
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	found := false
	for _, port := range kubernetesService.Spec.Ports {
		if port.Port == 53 && port.TargetPort.IntVal == dnsPort && port.Protocol == kapi.ProtocolTCP {
			found = true
		}
	}
	if !found {
		t.Fatalf("did not find DNS port in kubernetes service: %#v", kubernetesService)
	}
```

File: pkg/cmd/server/kubernetes/master_config.go [hunk 1]
```diff
@@ -229,8 +229,8 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 			return nil, fmt.Errorf("invalid DNS port: %v", err)
 		}
 		m.ExtraServicePorts = append(m.ExtraServicePorts,
-			kapi.ServicePort{Name: "dns", Port: dnsPort, Protocol: kapi.ProtocolUDP, TargetPort: util.NewIntOrStringFromInt(dnsPort)},
-			kapi.ServicePort{Name: "dns-tcp", Port: dnsPort, Protocol: kapi.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(dnsPort)},
+			kapi.ServicePort{Name: "dns", Port: 53, Protocol: kapi.ProtocolUDP, TargetPort: util.NewIntOrStringFromInt(53)},
```
Comment (liggitt) [inline pkg/cmd/server/kubernetes/master_config.go]: the target port still needs to be dnsPort

**Comment by abutcher** [comment]
@liggitt Oy, `targetPort` is now `dnsPort` and added the integration test.

**Comment by liggitt** [comment]
LGTM, [merge]

**Comment by abutcher** [comment]
@liggitt Test failures seem unrelated

**Comment by liggitt** [comment]
"Cancelled build" flake is fixed.
Image stream import fix is in process, some merges are still making it in
[test][merge]

**Pull request merged**